src/RemoveFile.cpp

```diff
@@ -0,0 +1,9 @@
+#include "RemoveFile.h"
+#include "NetMainThread.h"
+
+void RemoveFile::execute(void) {
+	NodeInfo * nInfo = NetMainThread::getNodeInfo();
+	// Check file owner
+	if(msg.firstField == nInfo->getOwnerId(msg.hash))
```

Comment:
I thought you'll be also checking if the message is from proper node (based on _nodeMap_ and IP from _in_addr_ after receive UDP InfoMessage) but it's also ok. Someone can spoof nodeId and delete file whose not owner, but preventing these situations is not essential now.